x/mint/abci_app.go

```diff
@@ -12,15 +12,15 @@ func BeginBlocker(ctx sdk.Context, k Keeper) {
 	params := k.GetParams(ctx)
 
 	// recalculate inflation rate
-	totalSupply := k.sk.TotalTokens(ctx)
+	stakingSupply := k.sk.StakingTokenSupply(ctx)
 	bondedRatio := k.sk.BondedRatio(ctx)
 	minter.Inflation = minter.NextInflationRate(params, bondedRatio)
-	minter.AnnualProvisions = minter.NextAnnualProvisions(params, totalSupply)
+	minter.AnnualProvisions = minter.NextAnnualProvisions(params, stakingSupply)
 	k.SetMinter(ctx, minter)
 
 	// mint coins, add to collected fees, update supply
 	mintedCoin := minter.BlockProvision(params)
 	k.fck.AddCollectedFees(ctx, sdk.Coins{mintedCoin})
-	k.sk.InflateSupply(ctx, mintedCoin.Amount)
+	k.sk.InflateNotBondedTokenSupply(ctx, mintedCoin.Amount)
```

Comment:
```suggestion
	k.sk.InflateUnbondedTokenSupply(ctx, mintedCoin.Amount)
```

Reply:
Also, should the supply keeper keep track of bonded/unbonded supply and the staking keeper have the supply keeper passed in?